cmd/destroy/cfghandler.go

```diff
@@ -31,6 +31,7 @@ type configMapHandler interface {
 	translateConfigMapAndDeploy(context.Context, *pipeline.CfgData) (*apiv1.ConfigMap, error)
 	destroyConfigMap(context.Context, *apiv1.ConfigMap, string) error
 	setErrorStatus(context.Context, *apiv1.ConfigMap, *pipeline.CfgData, error) error
+	getConfigmap(ctx context.Context, name, namespace string) (*apiv1.ConfigMap, error)
```

Comment:
I'd go with getting directly the variables, that way we abstract the get variables instead of retrieving the cmap and the if exists retrieving the vars. Same way we translate and deploy or set the error status. I don't like the fact that the deploy function needs to know that `getcmap` can return `nil, nil` WDYT?